### PR TITLE
fuzzy adding components

### PR DESCRIPTION
**Pull request by nelsonpecora**
* [trello ticket](https://trello.com/c/n7cR8jyg/27-add-one-off-components)
* allows adding ANY available component if allowd by a component list
* depends on amphora PR https://github.com/nymag/amphora/pull/305
* adds `all-components` service to make the code DRYer

### CODE REVIEW

File: behaviors/README.md [hunk 1]
```diff
@@ -197,6 +197,28 @@ footerMetadata:
 
 Like the Page Area property, this only works in a layout's component lists.
 
+#### Fuzzy Component Lists and Properties
+
+By default, component lists and properties allow strict whitelisting and blacklisting with `include` and `exclude`. This is appropriate for most situations, but there are certain scenarios where you might want to be less strict. For example, the body of an article might be paragraphs and images 99% of the time, but you still want to allow your interactives team to create one-off components and add them to articles.
+
+You can accomplish this by adding `fuzzy: true` to your component list / property. This will add a "View all components" button to the bottom of the component list in the add-components pane. When a user clicks the button, it'll open a different add-components pane with a list of all available components.
```
Comment (cruzanmo) [inline behaviors/README.md]: This sounds like a way to hide the component as an option? Maybe `hidden: true`?

Comment (nelsonpecora) [inline behaviors/README.md]: No, it makes the white/black list less strict by allowing users to (with an extra action) add _any_ component available in the clay instance. This allows for things like one-off interactives in articles (without having to manually to api calls against the server to add them)

Comment (cruzanmo) [inline behaviors/README.md]: Oh, I understand, that makes sense.

What about calling it `allowAll` or `allowAny`?

Comment (nelsonpecora) [inline behaviors/README.md]: we were debating that, but the discussion naturally settled on using `fuzzy` to talk about these kinds of lists

Comment (cruzanmo) [inline behaviors/README.md]: hmm, not my first choice, but if the team thinks it makes sense, that's cool.

**Comment by jonwinton** [comment]
👍

**Pull request merged**